ladon.go

```diff
@@ -48,13 +56,23 @@ func (l *Ladon) IsAllowed(r *Request) (err error) {
 // The IsAllowed interface should be preferred since it uses the manager directly. This is a lower level interface for when you don't want to use the ladon manager.
 func (l *Ladon) DoPoliciesAllow(r *Request, policies []Policy) (err error) {
 	var allowed = false
+	var deciders = Policies{}
+
+	defer func() {
+		if allowed {
+			l.auditLogger().OnGrantedAuthorization(r, policies, deciders)
```

Comment:
What would be nice here is if we could somehow log precedence as well. Let's take the following use case:

* Found policy 1, result: allowed
* Found policy 2, result: allowed
* Found policy 3, result: forcefully denied (e.g. banned user)

It would be nice to be able to see something like `Policies 1 and 2 allow access, but 3 forcefully denied the access request.`

I think this is already, sort of, possible because deciders is all the matched policies, but we would have to do some custom logic in the handler (although it wouldn't be too hard).

Not sure, what are your thoughts here?

Reply:
I think it's already possible with the following logic:
* if granted, we output `{deciders} allow access`
* if denied, we have:
    * if `len(decidiers) == len(candidates)`, we output `no policies granted acces`
    * else we output `{deciders[:-1]} allow access, but {deciders[-1]} forcefully denied`

Reply:
> if len(decidiers) == len(candidates), we output no policies granted acces

I think it will be `if len(candidates) == 0`

Everything else makes sense!

Reply:
By the way, could we add an example logger (like NoOp) for the stdlib `log` interface that does this logic?